exasol/language_container_activation.py

```diff
@@ -1,58 +1,90 @@
 from typing import Dict
-import re
 
 from exasol.secret_store import Secrets
+from exasol.connections import open_pyexasol_connection
 
 # All secret store entries with language container activation commands
 # will have this common prefix in their keys.
 ACTIVATION_KEY_PREFIX = "language_container_activation_"
 
 
+def get_registered_languages_string(conf: Secrets) -> str:
+    """
+    Gets the session level language definitions from the database. Returns a string
+    in a format of a space separated list: "<alias1>=<url1> <alias2=<url2> ..."
+    """
+
+    with open_pyexasol_connection(conf) as pyexasol_conn:
+        query_result = pyexasol_conn.execute(
+            "SELECT SESSION_VALUE FROM SYS.EXA_PARAMETERS WHERE PARAMETER_NAME='SCRIPT_LANGUAGES'"
+        ).fetchall()
+        return query_result[0][0]
+
+
+def get_registered_languages(conf: Secrets) -> Dict[str, str]:
+    """
+    Collects the existing, session level, language definitions from the database.
+    Returns them as a dictionary {alias: language_url}
+    """
+
+    lang_definitions_str = get_registered_languages_string(conf)
+    result: Dict[str, str] = {}
+    for lang_definition in lang_definitions_str.split():
+        alias, lang_url = lang_definition.split("=", maxsplit=1)
+        result[alias] = lang_url
+    return result
+
+
+def get_requested_languages(conf: Secrets) -> Dict[str, str]:
+    """
+    Collects language definitions from the secret store. Returns them as a
+    dictionary {alias: language_url}.
+
+    Raises a RuntimeError if tt finds two entries with the same alias but
+    different language definitions (i.e. URLs).
+    """
+
+    result: Dict[str, str] = {}
+    # Iterate over all entries that look like language definitions.
+    for key, value in conf.items():
+        if key.startswith(ACTIVATION_KEY_PREFIX):
+            alias, lang_url = value.split("=", maxsplit=1)
+            if alias in result:
+                if result[alias].casefold() != lang_url.casefold():
```

Comment:
the urls are actually not case in insensitive, because they depend on the file system and the Linux file systems are case-sensitive